packages/office-ui-fabric-react/src/components/Shimmer/examples/Shimmer.Application.Example.tsx

```diff
@@ -31,140 +41,91 @@ const fileIcons: { name: string }[] = [
+  const { current: state } = React.useRef<IShimmerApplicationExampleState>({
+    lastIntervalId: 0,
+    visibleCount: 0,
+  });
+
+  const [items, setItems] = React.useState<(IExampleItem | null)[] | undefined>(undefined);
+
+  const buildShimmerColumns: IColumn[] = useConst(() => {
+    const currentItems = createListItems(1);
+    const columns: IColumn[] = buildColumns(currentItems);
+    for (const column of columns) {
+      if (column.key === 'thumbnail') {
+        column.name = 'FileType';
+        column.minWidth = 16;
+        column.maxWidth = 16;
+        column.isIconOnly = true;
+        column.iconName = 'Page';
+        break;
+      }
+    }
+    return columns;
+  });
 
-  public render(): JSX.Element {
-    const { items, columns, isDataLoaded } = this.state;
+  const { setInterval, clearInterval } = useSetInterval();
 
-    return (
-      <div>
-        <Toggle
-          label="Toggle to load content"
-          style={{ display: 'block', marginBottom: '20px' }}
-          checked={isDataLoaded}
-          onChange={this._onLoadData}
-          onText="Content"
-          offText="Shimmer"
-        />
-        <div>
-          <ShimmeredDetailsList
-            setKey="items"
-            items={items}
-            columns={columns}
-            selectionMode={SelectionMode.none}
-            onRenderItemColumn={this._onRenderItemColumn}
-            enableShimmer={!isDataLoaded}
-            ariaLabelForShimmer="Content is being fetched"
-            ariaLabelForGrid="Item details"
-            listProps={{ renderedWindowsAhead: 0, renderedWindowsBehind: 0 }}
-          />
-        </div>
-      </div>
-    );
-  }
+  const loadMoreItems = (): void => {
+    state.visibleCount = Math.min(exampleItems.length, state.visibleCount + 2);
 
-  private _loadData = (): void => {
-    this._lastIntervalId = this._async.setInterval(() => {
-      const randomQuantity: number = Math.floor(Math.random() * 10) + 1;
-      const itemsCopy = this.state.items!.slice(0);
-      itemsCopy.splice(
-        this._lastIndexWithData,
-        randomQuantity,
-        ..._items.slice(this._lastIndexWithData, this._lastIndexWithData + randomQuantity),
-      );
-      this._lastIndexWithData += randomQuantity;
-      this.setState({
-        items: itemsCopy,
-      });
-    }, INTERVAL_DELAY);
+    setItems(exampleItems.map((current, index) => (index < state.visibleCount ? current : null)) as IExampleItem[]);
   };
 
-  private _onLoadData = (ev: React.MouseEvent<HTMLElement>, checked: boolean): void => {
-    if (!_items) {
-      _items = createListItems(ITEMS_COUNT);
-      _items.forEach((item: IExampleItem) => {
-        const randomFileType = this._randomFileIcon();
-        item.thumbnail = randomFileType.url;
-      });
-    }
-
-    let items: IExampleItem[];
-    const randomQuantity: number = Math.floor(Math.random() * 10) + 1;
+  const onLoadData = React.useCallback((ev: React.MouseEvent<HTMLElement>, checked: boolean): void => {
```

Comment:
Best to do `useConstCallback` here since you don't care about dep changes. (Also, that should prevent it from getting flagged by the "rules of hooks" eslint rule I'm about to enable for not having exhaustive deps.)